worker/src/handlers/schedule.ts

```diff
@@ -1,181 +1,208 @@
-    ...lastDataEntry,
-    active_installations: storedData,
+  await prosessQueue();
+}
+
+async function prosessQueue(): Promise<void> {
+  const queue = (await KV.get<Queue>("queue", "json")) || {
+    entries: [],
+    data: {},
   };
 
-  //await KV.put(storeKey, JSON.stringify(currentDataset));
-  await KV.put("core_analytics", JSON.stringify(core_analytics));
+  if (queue.entries.length === 0) {
+    // No entries, get list
+    console.log("No entries, get list");
+    queue.entries = await listKV("uuid");
+  }
+
+  // Prosess the first 850 entries in the array
+  for (const entryKey of queue.entries.slice(0, 850)) {
+    let entryData;
+    try {
+      entryData = await KV.get<SanitizedPayload>(entryKey, "json");
+    } catch (e) {
+      console.log(e);
+    }
+
+    if (entryData !== undefined && entryData !== null) {
+      queue.data = combineEntryData(queue.data, entryData);
+    }
+  }
+  queue.entries = queue.entries.slice(850);
+
+  if (queue.entries.length === 0) {
+    // No more entries, store and reset queue data
+    console.log("No more entries, store and reset queue data");
+    const core_analytics: Record<string, any> = {};
+    const timestampString = String(new Date().getTime());
+
+    const queue_data = processQueueData(queue.data as QueueData);
+    const storedAnalytics =
+      (await KV.get<{ [key: string]: CurrentAnalytics }>(
+        "core_analytics",
+        "json"
+      )) || {};
+
+    for (const key of Object.keys(storedAnalytics)) {
+      core_analytics[key] = {
+        active_installations: storedAnalytics[key].active_installations,
+        installation_types: storedAnalytics[key].installation_types,
+      };
+    }
+
+    core_analytics[timestampString] = queue_data;
+
+    await KV.put(`history:${timestampString}`, JSON.stringify(queue_data));
+    await KV.put("core_analytics", JSON.stringify(core_analytics));
+    queue.data = {};
+  }
+
+  await KV.put("queue", JSON.stringify(queue));
 }
 
-export async function listStoredData(): Promise<number> {
-  const uuidList: Set<string> = new Set();
-  const uuidData: SanitizedPayload[] = [];
-  let entries = 0;
+async function listKV(prefix: string): Promise<string[]> {
+  let entries: Set<string> = new Set();
 
   let lastResponse;
   while (lastResponse === undefined || !lastResponse.list_complete) {
     lastResponse = await KV.list({
-      prefix: "uuid",
+      prefix,
       cursor: lastResponse !== undefined ? lastResponse.cursor : undefined,
     });
 
-    entries = entries + lastResponse.keys.length;
-
-    /*
     for (const key of lastResponse.keys) {
-      uuidList.add(key.name);
+      entries.add(key.name);
     }
-    */
   }
 
-  return entries;
+  return Array.from(entries);
+}
 
-  /*  for (const storageKey of uuidList) {
-    const payload = await KV.get<SanitizedPayload>(storageKey, "json");
-    if (payload) {
-      uuidData.push(payload);
-    }
+function combineEntryData(
+  data: Partial<QueueData>,
```

Comment:
Why not just always require a full `QueueData`? And just have a function to create empty queue data object.